apm-agent-plugins/apm-quartz-job-plugin/src/test/java/co/elastic/apm/agent/quartzjob/JobTransactionNameInstrumentationTest.java

```diff
@@ -135,51 +124,76 @@ public void testAgentPaused() throws SchedulerException {
     @Test
     void testSpringJob() throws SchedulerException {
         JobDetail job = JobBuilder.newJob(TestSpringJob.class)
-            .withIdentity("dummyJobName", "group1").build();
-        Trigger trigger = TriggerBuilder
-            .newTrigger()
-            .withIdentity("myTrigger")
-            .withSchedule(
-                SimpleScheduleBuilder.repeatSecondlyForTotalCount(1, 1))
+            .withIdentity("dummyJobName", "group1")
             .build();
-        scheduler.scheduleJob(job, trigger);
-        verifyJobDetails(job);
+        scheduler.scheduleJob(job, createTrigger());
+
+        verifyTransactionFromJobDetails(job);
     }
 
     @Test
     void testJobWithResult() throws SchedulerException {
         JobDetail job = JobBuilder.newJob(TestJobWithResult.class)
-            .withIdentity("dummyJobName").build();
-        Trigger trigger = TriggerBuilder
-            .newTrigger()
-            .withIdentity("myTrigger")
-            .withSchedule(
-                SimpleScheduleBuilder.repeatSecondlyForTotalCount(1, 1))
+            .withIdentity("dummyJobName")
             .build();
-        scheduler.scheduleJob(job, trigger);
-        verifyJobDetails(job);
-        assertThat(reporter.getTransactions().get(0).getResult()).isEqualTo("this is the result");
+        scheduler.scheduleJob(job, createTrigger());
+
+        Transaction transaction = verifyTransactionFromJobDetails(job);
+        assertThat(transaction.getResult()).isEqualTo("this is the result");
     }
 
     @Test
     void testDirectoryScan() throws SchedulerException, IOException {
         Path directoryScanTest = Files.createTempDirectory("DirectoryScanTest");
 
-        Trigger trigger = TriggerBuilder
-            .newTrigger()
-            .withIdentity("myTrigger")
-            .withSchedule(
-                SimpleScheduleBuilder.repeatSecondlyForTotalCount(1, 1))
-            .build();
         final JobDetail job = JobBuilder.newJob(DirectoryScanJob.class)
             .withIdentity("dummyJobName")
             .usingJobData(DirectoryScanJob.DIRECTORY_NAME, directoryScanTest.toAbsolutePath().toString())
             .usingJobData(DirectoryScanJob.DIRECTORY_SCAN_LISTENER_NAME, TestDirectoryScanListener.class.getSimpleName())
             .build();
 
         scheduler.getContext().put(TestDirectoryScanListener.class.getSimpleName(), new TestDirectoryScanListener());
-        scheduler.scheduleJob(job, trigger);
-        verifyJobDetails(job);
+        scheduler.scheduleJob(job, createTrigger());
+
+        verifyTransactionFromJobDetails(job);
+    }
+
+    @Test
+    void testJobWithException() throws SchedulerException {
+        JobDetail job = JobBuilder.newJob(TestJobWithException.class).withIdentity("dummyJobName").build();
+        scheduler.scheduleJob(job, createTrigger());
+
+        Transaction transaction = verifyTransactionFromJobDetails(job);
+        assertThat(transaction.getOutcome()).isEqualTo(Outcome.FAILURE);
```

Comment:
What about verifying that the other are successes?
You can move this check to `verifyTransactionFromJobDetails()` to compare to a provided expected outcome.